### PR TITLE
Add new initial fields to v20240812preview

### PR DESCRIPTION
**Pull request by cadenmarchese**
### Which issue this PR addresses:

Fixes https://issues.redhat.com/browse/ARO-4381 && https://issues.redhat.com/browse/ARO-4377, unblocks https://issues.redhat.com/browse/ARO-3979

### What this PR does / why we need it:

This is the first of 3-ish PRs to add new fields to the preview API + converter. There will be additional fields added to support OCP upgrades as well as Cluster MSI.

### CODE REVIEW

File: pkg/api/v20240812preview/openshiftcluster.go [hunk 1]
```diff
@@ -55,6 +58,9 @@ type OpenShiftClusterProperties struct {
 	// The cluster service principal profile.
 	ServicePrincipalProfile ServicePrincipalProfile `json:"servicePrincipalProfile,omitempty"`
 
+	// The workload identity profile.
+	PlatformWorkloadIdentityProfile PlatformWorkloadIdentityProfile `json:"platformWorkloadIdentityProfile,omitempty"`
```
Comment (kimorris27) [inline pkg/api/v20240812preview/openshiftcluster.go:62]: This field is a pointer in the design doc to allow for it to be `nil` for CSP clusters.

File: pkg/api/v20240812preview/openshiftcluster.go [hunk 2]
```diff
@@ -276,6 +288,33 @@ type IngressProfile struct {
 	IP string `json:"ip,omitempty" swagger:"readOnly"`
 }
 
+type PlatformWorkloadIdentityProfile struct {
+	PlatformWorkloadIdentities []PlatformWorkloadIdentity `json:"platformWorkloadIdentities,omitempty"`
+}
+
+// PlatformWorkloadIdentity stores information representing a single workload identity.
+type PlatformWorkloadIdentity struct {
+	Name       string `json:"name,omitempty"`
```
Comment (kimorris27) [inline pkg/api/v20240812preview/openshiftcluster.go:297]: This was originally `Name`, but we later changed it to `OperatorName` in the design doc to make the purpose of the field more clear.

File: pkg/api/openshiftcluster.go [hunk 3]
```diff
@@ -122,6 +123,8 @@ type OpenShiftClusterProperties struct {
 
 	ServicePrincipalProfile ServicePrincipalProfile `json:"servicePrincipalProfile,omitempty"`
 
+	PlatformWorkloadIdentityProfile PlatformWorkloadIdentityProfile `json:"platformWorkloadIdentityProfile,omitempty"`
```
Comment (kimorris27) [inline pkg/api/openshiftcluster.go:126]: This field is a pointer in the design doc to allow for it to be `nil` for CSP clusters.

File: pkg/api/openshiftcluster.go [hunk 4]
```diff
@@ -761,3 +767,38 @@ type HiveProfile struct {
...
+	MissingFields
+
+	PlatformWorkloadIdentities []PlatformWorkloadIdentity `json:"platformWorkloadIdentities,omitempty"`
+}
+
+// PlatformWorkloadIdentity stores information representing a single workload identity.
+type PlatformWorkloadIdentity struct {
+	MissingFields
+
+	Name       string `json:"name,omitempty"`
```
Comment (kimorris27) [inline pkg/api/openshiftcluster.go:781]: This was originally `Name`, but we later changed it to `OperatorName` in the design doc to make the purpose of the field more clear.

**Review by kimorris27 — CHANGES_REQUESTED** [review]
I suggested some small changes to make sure we're lining up with what the team agreed on in the design doc.

**Review by kimorris27 — APPROVED** [review]

**Review by kimorris27 — CHANGES_REQUESTED** [review]
Just noticed that we're missing the admin API changes. Let's add those into this PR.

**Review by kimorris27 — DISMISSED** [review]

File: pkg/api/admin/openshiftcluster.go [hunk 5]
```diff
@@ -24,27 +24,29 @@ type OpenShiftCluster struct {
...
+	MaintenanceTask                 MaintenanceTask                  `json:"maintenanceTask,omitempty" mutable:"true"`
+	OperatorFlags                   OperatorFlags                    `json:"operatorFlags,omitempty" mutable:"true"`
+	OperatorVersion                 string                           `json:"operatorVersion,omitempty" mutable:"true"`
+	CreatedAt                       time.Time                        `json:"createdAt,omitempty"`
+	CreatedBy                       string                           `json:"createdBy,omitempty"`
+	ProvisionedBy                   string                           `json:"provisionedBy,omitempty"`
+	ClusterProfile                  ClusterProfile                   `json:"clusterProfile,omitempty"`
+	FeatureProfile                  FeatureProfile                   `json:"featureProfile,omitempty"`
+	ConsoleProfile                  ConsoleProfile                   `json:"consoleProfile,omitempty"`
+	ServicePrincipalProfile         ServicePrincipalProfile          `json:"servicePrincipalProfile,omitempty"`
```
Comment (tsatam) [inline pkg/api/admin/openshiftcluster.go:46]: Are we planning on turning this field into a pointer, to allow it to be nil for WIMI clusters (inverse of `PlatformWorkloadIdentityProfile` below)? 

Making this change would require us to ensure nil checks across all of our references to the field throughout the codebase, so I'm in favor of it being left out of this PR and in a dedicated one if we do go down that path.

**Review by tsatam — APPROVED** [review]
LGTM, I've left one question about handling the existing ServicePrincipalProfile struct on WIMI clusters for my own curiosity, but it shouldn't block this PR and it would likely need to be the subject of its own dedicated PR if we want to tackle it.

File: pkg/api/admin/openshiftcluster.go (hunk 5 above)
Comment (kimorris27) [reply pkg/api/admin/openshiftcluster.go:46]: I'm personally in favor of making that change, and I agree that it should go in a dedicated PR.

Comment (cadenmarchese) [reply pkg/api/admin/openshiftcluster.go:46]: Agreed, we have talked about doing this as well. I created a dedicated card for this: https://issues.redhat.com/browse/ARO-6623

File: pkg/api/admin/openshiftcluster.go [hunk 6]
```diff
@@ -409,6 +415,33 @@ type IngressProfile struct {
 	IP         string     `json:"ip,omitempty"`
 }
 
+type PlatformWorkloadIdentityProfile struct {
```
Comment (s-amann) [inline pkg/api/admin/openshiftcluster.go:418]: should we also add some doc comments here?

File: pkg/api/openshiftcluster.go [hunk 7]
```diff
@@ -248,6 +251,8 @@ const (
 	FipsValidatedModulesDisabled FipsValidatedModules = "Disabled"
 )
 
+type OIDCIssuer string
```
Comment (s-amann) [inline pkg/api/openshiftcluster.go:254]: we have doc comments in the admin version, should we also add some doc comments here?

File: pkg/api/openshiftcluster.go [hunk 8]
```diff
@@ -761,3 +767,38 @@ type HiveProfile struct {
 	// ClusterDeployments.
 	CreatedByHive bool `json:"createdByHive,omitempty"`
 }
+
+type PlatformWorkloadIdentityProfile struct {
```
Comment (s-amann) [inline pkg/api/openshiftcluster.go:771]: should we also add some doc comments here?

File: pkg/api/v20240812preview/openshiftcluster.go [hunk 9]
```diff
@@ -276,6 +288,33 @@ type IngressProfile struct {
 	IP string `json:"ip,omitempty" swagger:"readOnly"`
 }
 
+type PlatformWorkloadIdentityProfile struct {
```
Comment (s-amann) [inline pkg/api/v20240812preview/openshiftcluster.go:291]: should we also add some doc comments here?

**Review by s-amann — APPROVED** [review]
lgtm. Some minor comments below which could be in a new PR

**Comment by cadenmarchese** [comment]
@s-amann should be good now, thanks for pointing that out!

**Review by kimorris27 — APPROVED** [review]

**Review by kimorris27 — CHANGES_REQUESTED** [review]
I checked out the CI failure, and it looks like the full results of `make client` haven't yet been pushed since the most recent changes.

**Review by tsatam — APPROVED** [review]

**Review by s-amann — APPROVED** [review]

**Review by kimorris27 — APPROVED** [review]

**Pull request merged**